src/components/Graph/Graph.tsx

```diff
@@ -41,11 +32,12 @@ const Graph: FC<GraphProps> = ({
 
   const setPath = useCallback(
     (value: PathwayResults): void => {
-      _setPath(value.path);
+      if (!(value.path.toString() === path.toString())) _setPath(value.path);
```

Comment:
I think using `!==` more clearly conveys what you're trying to do and is a little simpler.

```suggestion
      if (value.path.toString() !== path.toString()) _setPath(value.path);
```

Also why was this additional logic needed?

Reply:
Because setPath doesn't just set the path, and it gets called while the value.path is the same but it still causes the component to rerender.  But we still might need the second part of setPath.

Reply:
Ah ok thanks that explains it!  I was thinking that the component wouldn't rerender if a `React.setState` call if the resulting state was the same, but after doing a little reading based on your comment it appears components unconditionally re-render after every setState call regardless

Reply:
Any thoughts on the suggested code change?

Reply:
the suggested change seems fine to me

Reply:
Also I wonder if we could be stepping on a landmine doing this because setState is called asynchronously (so the previous state we're comparing against might not be correct).

React provides a method of doing updates computed against the previous state for this using functional updates: https://reactjs.org/docs/hooks-reference.html#functional-updates

Reply:
@KeeyanGhoreshi implemented a better solution here that moves the conditional check into a functional state update.

Reply:
@KeeyanGhoreshi after the rebase I think this updated code was lost.  I don't see the original if statement added or the functional state update...